Monacs.Core/Option.cs

```diff
@@ -1,21 +1,40 @@
-﻿using System;
+using System;
 using System.Collections.Generic;
-using System.Linq;
 
 namespace Monacs.Core
 {
+    /// <summary>
+    /// Encapsulates optional value. A value either contains a value of a type <see cref="T"/> or it is default of <see cref="T"/>.
+    /// <para />An Option becomes Some if any value is provided. Otherwise it's None.
+    /// </summary>
+    /// <typeparam name="T">Type of optional value.</typeparam>
     public struct Option<T> : IEquatable<Option<T>>, IEquatable<T>
     {
         internal Option(T value)
         {
             Value = value;
             IsSome = true;
         }
-
+        
+        /// <summary>
+        /// Encapsulated valid value.
+        /// <para /> !!! It is not recommended to use it directly !!!
+        /// <para /> Use one of the following extension methods instead:
+        /// <para /> * <see cref="Option.GetOrDefault{T}"/>, 
+        /// <para /> * <see cref="Option.Map{TIn,TOut}"/>, 
+        /// <para /> * <see cref="Option.Bind{TIn,TOut}"/>,
+        /// <para /> * <see cref="Option.Match{TIn,TOut}"/> 
+        /// </summary>
         public T Value { get; }
 
+        /// <summary>
+        /// Returns true if the option has a value that is not None.
+        /// </summary>
         public bool IsSome { get; }
 
+        /// <summary>
+        /// Returns true if the option has the None value.

```

Comment:
`if the Option has no value` or `if the Option is None` sounds better

Reply:
I will change to snd example